src/myARMSim.c

```diff
@@ -60,19 +62,19 @@ void run_armsim() {
 // it is used to set the reset values
 //reset all registers and memory content to 0
 void reset_proc() {
-  int i;
-  for(i=0;i<16;i++)
+  for(int i=0; i<16; i++){
```

Comment:
This code is written in C. in loop declaration of variables won't work.
Also it isn't C99